core/state_transition.go

```diff
@@ -634,10 +640,10 @@ func (st *stateTransition) innerExecute() (*ExecutionResult, error) {
 	if st.msg.IsDepositTx && rules.IsOptimismRegolith {
 		// Skip coinbase payments for deposit tx in Regolith
 		return &ExecutionResult{
-			UsedGas:     st.gasUsed(),
-			RefundedGas: gasRefund,
-			Err:         vmerr,
-			ReturnData:  ret,
+			UsedGas:    st.gasUsed(),
+			MaxUsedGas: peakGasUsed,
```

Comment:
See other comment, the only place where the `MaxUsedGas` from the `ExecutionResult` is used is in the gas estimator at `eth/gasestimator/gasestimator.go` - and this is irrelevant to deposit txs. So we could to not set it here at all, to be consistent with the pre-Regolith case. But it doesn't hurt either...